warehouse-management-service/main.go

```diff
@@ -4,45 +4,57 @@ import (
 	"fmt"
 	"net/http"
 	"os"
+	"warehouse-management-service/internal/config"
+	"warehouse-management-service/internal/handler"
+	"warehouse-management-service/pkg/database/postgres"
 
-	"github.com/gin-gonic/gin"
+	_ "github.com/lib/pq"
+)
+
+const (
+	CONFIG_FILE_PATH_ENV_VARIABLE = "CONFIG_FILE_PATH"
 )
 
 func main() {
-	err := initialize()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "App startup error: %v", err)
-		os.Exit(1)
+	// Get absolute path of config file from env variable
+	configFilePath, ok := os.LookupEnv(CONFIG_FILE_PATH_ENV_VARIABLE)
+	if !ok {
+		fmt.Fprintf(os.Stderr, "%s env variable not set", CONFIG_FILE_PATH_ENV_VARIABLE)
 	}
-}
 
-func initialize() error {
-	err := setupDatabase()
+	// Get config
+	config, err := config.FromFile(configFilePath)
 	if err != nil {
-		return err
+		fmt.Fprintf(os.Stderr, "App startup error: %v", err)
+		os.Exit(1)
 	}
 
-	err = startServer()
+	// Get Postgres DB service
+	db, err := postgres.New(config.Postgres.Host,
+		config.Postgres.Port,
+		config.Postgres.Username,
+		config.Postgres.Password,
+		config.Postgres.DBName,
+		config.Postgres.SSLMode)
 	if err != nil {
-		return err
+		fmt.Fprintf(os.Stderr, "App startup error: %v", err)
+		os.Exit(1)
 	}
 
-	return nil
-}
-
-func setupDatabase() error {
-	// TODO
-	return nil
-}
+	// Close DB connection at shutdown
+	defer func() {
+		err = db.Close()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+		}
+	}()
 
-func startServer() error {
-	router := gin.Default()
+	handler := handler.New(db)
 
-	router.GET("/ping", func(ctx *gin.Context) {
-		ctx.JSON(http.StatusOK, gin.H{
-			"message": "pong",
-		})
-	})
+	http.ListenAndServe(":80", handler)
```

Comment:
issue: We are ignoring the error value here, and so the subsequent err != nil check is reading the outdated `err` value

Read this into `err` to fix this:
```suggestion
	err = http.ListenAndServe(":80", handler)
```

Reply:
note: Use a linter to avoid these slips in the future. I actually found this one out by running `golangci-lint` with the default settings. Better yet, add the lint job to the CI.

Reply:
Fixed this.

About adding this to CI, I've taken a note of this. I'll incorporate it into CI when I set it up.